test/modules/landing/landing_presenter_test.dart

```diff
@@ -0,0 +1,107 @@
+import 'package:mockito/annotations.dart';
+import 'package:mockito/mockito.dart';
+import 'package:quick_test/quick_test.dart';
+import 'package:streams_provider/streams_provider.dart';
+import 'package:survey/modules/landing/landing_module.dart';
+import 'package:survey/services/locator/locator_service.dart';
+
+import '../../mocks/build_context.dart';
+import 'landing_presenter_test.mocks.dart';
+
+@GenerateMocks(
+    [LandingView, LandingRouter, LandingInteractor, StreamsDisposeBag])
+void main() {
+  describe("a Landing presenter", () {
+    late LandingPresenterImpl presenter;
+    late MockLandingView view;
+    late MockLandingRouter router;
+    late MockLandingInteractor interactor;
+    late MockBuildContext buildContext;
+    late MockStreamsDisposeBag disposeBag;
+
+    beforeEach(() {
+      disposeBag = MockStreamsDisposeBag();
+      locator.registerFactory<StreamsDisposeBag>(() => disposeBag);
+
+      buildContext = MockBuildContext();
+
+      view = MockLandingView();
+      when(view.context).thenReturn(buildContext);
```

Comment:
question: if we're testing presenter, a View's real implementation won't be necessary to be initialized no? 🤔  not even with MockBuildContext 🤔  or am I missing something?

Reply:
@sleepylee we don't need to initialize a real view there, cause we only need to verify presenter call the right method from the view's interface.
I need  this line `when(view.context).thenReturn(buildContext);` because `router` will need that 😁